lockfile/src/parsers/pypi.rs

```diff
@@ -52,10 +53,22 @@ fn line(input: &str) -> IResult<&str, &str> {
         line = "";
     }
 
+    // Ignore index config options.
+    //
+    // Since `ThirdPartyVersion` only allows a single registry, we only record the
+    // primary one.
+    if let Some(index_url) = line.strip_prefix("--index-url") {
```

Comment:
The `--index-url` option can also be specified as `-i`.
Granted, for `pip-compile` generated files, the long option name will be used, but these requirements files are often hand generated as well.